frontend/src/app/services/event-list.ts

```diff
@@ -4,8 +4,43 @@ import {dateDiff} from './utils';
 import {formatNumber} from '../utils/format';
 
 function seasonalDateDiff(date: string, mmdd: string) {
```

Comment:
The concept of this function is not understandable/practical. It seems to be intended to compute the distance in days, similar as `getSeasonality`. Check these, just 2 days one after another. Please, rethink/improve.

```typescript
seasonalDateDiff('2020-01-01', '12-31'); // -365
seasonalDateDiff('2020-12-31', '01-01'); // 365
```

Reply:
The result should be between -183 and +183. If the first dateDiff is too big in absolute value, we need to adjust the year by 1.

Reply:
Does before and after make any difference? If not, I would `abs` it and name it `seasonalDateDistance`.

Reply:
We allow both dates without year (for example warm season start `05-20` that will occur every year) and with year (`2021-01-05`) to be used. That's why I'm guessing the year in `dateWithYear`.
I think it causes much more confusion than if I forced all dates to be explicit with year - even if we'd have to duplicate the yearly repeated events.

The warm season start may be repeated by using the `timeout` parameter; possibly:
- `timeout: 365` - summer in 2021 starts at the exact day as the summer in 2020
- `timeout: _.random(353, 377)` - summer in 2021 might start in interval (05-12 ... 06-26) (not uniform probability) instead of correct(05-20 ... 06-14)
- (pseudo) `timeout: 365 - dateDiff(s.date, '2020-06-02')`  (06-02 being the middle of the interval). This seems correct but doesn't feel very readable (and contains a data dependency between that interval middle and interval bounds )

I will go with requiring year in date and solve the warm season with that randomized timout for now.

Reply:
I modified the code to always use the full date with year. That allowed me to remove most of the date handling functions.